src/python/pants/core/goals/update_build_files.py

```diff
@@ -556,24 +556,23 @@ def next_token_is(string: str, token_type=tokenize.OP) -> bool:
             return next_token.type is token_type and next_token.string == string
         return False
 
-    def should_be_renamed(token: tokenize.TokenInfo) -> bool:
+    def is_candidate_field(token: tokenize.TokenInfo) -> bool:
+        """Return true if the token is a field and belongs to a target with renames."""
         nonlocal pants_target
-
         if not parse_target(token):
             return False
-
-        if pants_target not in renamed_field_types.target_field_renames:
-            return False
-
-        return (
-            next_token_is("=")
-            and token.string in renamed_field_types.target_field_renames[pants_target]
-        )
+        return pants_target in renamed_field_types.target_field_renames and next_token_is("=")
 
     updated_text_lines = list(request.lines)
     for token in tokens:
-        if not should_be_renamed(token):
+        if not is_candidate_field(token):
+            continue
+        if token.string == "overrides":
+            # TODO. We can only worry about `{}` syntax for now, not `dict()`.
             continue
```

Comment:
@kaos any tips for how to approach this code? The tricky thing is that we are not actually renaming the overrides token, we are going to rename a value nested inside of it. I'm thinking that we will need to keep track with a variable when we are inside an overrides block or not. Maybe that `level` variable.